kotlin/services/eventbridge/src/main/kotlin/com/kotlin/eventbridge/EventbridgeMVP.kt

```diff
@@ -0,0 +1,729 @@
+        snsClient.deleteTopic(request)
+        println(" $topicArnVal was deleted.")
+    }
+}
+
+// snippet-start:[eventbridge.kotlin._delete_rule.main]
+suspend fun deleteRuleByName(ruleName: String?) {
+    val ruleRequest = DeleteRuleRequest {
+        name = ruleName
+    }
+    EventBridgeClient { region = "us-east-1" }.use { eventBrClient ->
+        eventBrClient.deleteRule(ruleRequest)
+        println("Successfully deleted the rule")
+    }
+}
+// snippet-end:[eventbridge.kotlin._delete_rule.main]
+
+// snippet-start:[eventbridge.kotlin.delete.targets.main]
+suspend fun deleteTargetsFromRule(eventRuleName: String?) {
+    // First, get all targets that will be deleted.
+    val request = ListTargetsByRuleRequest {
+        rule = eventRuleName
+    }
+
+    EventBridgeClient { region = "us-east-1" }.use { eventBrClient ->
+        val response = eventBrClient.listTargetsByRule(request)
+        val allTargets = response.targets
+
+        // Get all targets and delete them.
+        if (allTargets != null) {
+            for (myTarget in allTargets) {
+                val removeTargetsRequest = RemoveTargetsRequest {
+                    rule = eventRuleName
+                    ids = listOf(myTarget.id.toString())
+                }
+                eventBrClient.removeTargets(removeTargetsRequest)
+                println("Successfully removed the target")
+            }
+        }
+    }
+}
+// snippet-end:[eventbridge.kotlin.delete.targets.main]
+
+// snippet-start:[eventbridge.kotlin._put_event.main]
+suspend fun triggerCustomRule(email: String) {
+    val json = "{" +
+        "\"UserEmail\": \"" + email + "\"," +
+        "\"Message\": \"This event was generated by example code.\"" +
+        "\"UtcTime\": \"Now.\"" +
+        "}"
+
+    val entry = PutEventsRequestEntry {
+        source = "ExampleSource"
+        detail = json
+        detailType = "ExampleType"
+    }
+
+    val eventsRequest = PutEventsRequest {
+        this.entries = listOf(entry)
+    }
+
+    EventBridgeClient { region = "us-east-1" }.use { eventBrClient ->
+        eventBrClient.putEvents(eventsRequest)
+    }
+}
+// snippet-end:[eventbridge.kotlin._put_event.main]
+
+suspend fun updateCustomRuleTargetWithTransform(topicArn: String?, ruleName: String?) {
+    val targetId = UUID.randomUUID().toString()
+
+    val inputTransformerOb = InputTransformer {
+        inputTemplate = "\"Notification: sample event was received.\""
+    }
+
+    val target = aws.sdk.kotlin.services.eventbridge.model.Target {
+        id = targetId
+        arn = topicArn
+        inputTransformer = inputTransformerOb
+    }
+
+    val targetsRequest = PutTargetsRequest {
+        rule = ruleName
+        targets = listOf(target)
+        eventBusName = null
+    }
+
+    EventBridgeClient { region = "us-east-1" }.use { eventBrClient ->
+        eventBrClient.putTargets(targetsRequest)
+    }
+}
+
+// snippet-start:[eventbridge.kotlin.puttargetstransform.main]
```

Comment:
Is this tag correct? It looks like this should be under PutRule but it is showing up under PutTarget